src/test/java/spoon/test/template/TemplateTest.java

```diff
@@ -442,6 +439,25 @@ public void testCheckBoundTemplate() throws Exception {
 		assertEquals(injectedCode, method.getBody().getStatement(0));
 	}
 
+	@Test
+	public void testAbstractTemplateIsWellFormed() {
+		// contract: IsWellFormed method of AbstractTemplate class returns false if there are zero template parameter
+
+		// arrange
+		AbstractTemplate abstractTemplate = new AbstractTemplate() {
+			@Override
+			public CtElement apply(CtType targetType) {
+				return null;
+			}
+		};
+
+		// act
+		boolean zeroTemplateParameters = abstractTemplate.isWellFormed();
+
+		// assert
+		assertFalse(zeroTemplateParameters);
```

Comment:
This is a matter of taste, but I typically collapse the act and assert if they are simple enough, and there's only one of each. If you _want_ to separate them, that's also fine, but I think this would be equally viable here.

```suggestion
		assertFalse(abstractTemplate.isWellFormed());
```